bigquery/storage/managedwriter/adapt/error.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package adapt
+
+import "fmt"
+
+type ConversionError struct {
+	Location string
+	Details  error
+}
+
+func (ce *ConversionError) Error() string {
+	if ce.Location == "" {
+		return ce.Details.Error()
+	}
+	return fmt.Sprintf("Location: %s, Error: %v", ce.Location, ce.Details)
```

Comment:
nit: Start with a a lowercase letter and not need for the word error.
It also might be nice to make the location part more of a phrase. i.e.

```
fmt.Sprintf("unable to convert schema with location %q: %v", ce.Location, ce.Details)
```

Reply:
fmt.Sprintf("conversion error in location %q: %v", ce.Location, ce.Details)